apps/smart-contracts/token/test/AllowlistPurchaseHook.test.ts

```diff
@@ -45,6 +57,46 @@ describe('AllowlistPurchaseHook', () => {
     })
   })
 
+  describe('# hook', () => {
+    let purchaser: SignerWithAddress
+    let recipient: SignerWithAddress
+
+    beforeEach(async () => {
+      await setupHookAndList()
+      await allowlistPurchaseHook.connect(owner).setAllowedContract(allowedContract.address)
+      purchaser = user1
+      recipient = user2
+    })
+
+    it('reverts if caller is not allowed contract', async () => {
+      expect(await allowlistPurchaseHook.getAllowedContract()).to.not.eq(user1.address)
+
+      await expect(
+        allowlistPurchaseHook.connect(user1).hook(purchaser.address, recipient.address, 1, 1)
+      ).to.be.revertedWith('msg.sender != allowed contract')
+    })
+
+    it('reverts if recipient not allowed', async () => {
+      allowedAccounts.isIncluded.whenCalledWith(recipient.address).returns(false)
+
+      await expect(
+        allowlistPurchaseHook
+          .connect(allowedContract)
+          .hook(purchaser.address, recipient.address, 1, 1)
+      ).to.be.revertedWith('Recipient not allowed')
+    })
+
+    it('succeeds if recipient allowed', async () => {
+      allowedAccounts.isIncluded.whenCalledWith(recipient.address).returns(true)
+
+      await expect(
+        allowlistPurchaseHook
+          .connect(allowedContract)
+          .hook(purchaser.address, recipient.address, 1, 1)
+      ).to.not.reverted
+    })
```

Comment:
```suggestion
    it('reverts if recipient not allowed', async () => {
      allowedAccounts.isIncluded.whenCalledWith(recipient.address).returns(false)

      await expect(
        allowlistPurchaseHook
          .connect(allowedContract)
          .hook(purchaser.address, recipient.address, 1, 1)
      ).to.be.revertedWith('Recipient not allowed')
    })

    it('succeeds if recipient allowed', async () => {
      allowedAccounts.isIncluded.whenCalledWith(recipient.address).returns(true)

      await expect(
        allowlistPurchaseHook
          .connect(allowedContract)
          .hook(purchaser.address, recipient.address, 1, 1)
      ).to.not.reverted
    })
```

I think since we are using the smock library here, it is preferable that we simply verify that the `isIncluded` function is called with the correct parameters and that reverts are passed upward, rather than re-testing the underlying behavior.

Reply:
Nilay and I have concluded that we should just add another test case verifying that the function is called with the correct arguments. Since we still need these tests to ensure `Recipient not allowed` is thrown (this belongs to the Hook and not AccountList, so we need to include it in this test suite).